### PR TITLE
Feature/purchase routes

### PR DESCRIPTION
**Pull request by santiagonoguez**
## Description
Change /license route to /license/add and add /license/remove skeleton ( just title for now ). 

I removed most of the structure on the HTML remove page for now, to avoid confusions. 

[stage-9]

## Motivation and Context
Part of remove licenses requirement / discussion

## How Has This Been Tested?
The add license flow was tested from display page and for bulk licensing, and also page reload was tested. Both locally and on [stage-9]

Unit tests and one E2E test path were updated, and tests for remove routes were added, mirroring the ones for add.

## Release Plan:
This branch is intended to be used as base branch for remove licenses changes. It could also be merged immediately to master as the remove page aren't visible to end user until navigation from subscription page is added. We can discuss this further in next meeting.

#### Release Checklist Items Skipped?
No need to notify support.

### CODE REVIEW

File: web/partials/purchase/purchase-licenses-remove.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div class="madero-style">
+  <div class="app-header u_padding-20-vertical mb-5 border-bottom">
+    <h1 class="app-header-title">Remove Display Licenses</h1>
+  </div>
+
+  <div rv-spinner rv-spinner-key="purchase-licenses-loader" rv-spinner-start-active="1"></div>
```
Comment (santiagonoguez) [inline web/partials/purchase/purchase-licenses-remove.html:6]: should I use a different spinner key for this other page ?

File: web/scripts/purchase/app.js [hunk 2]
```diff
@@ -52,10 +59,8 @@ angular.module('risevision.apps')
         })
         .state('apps.purchase.licenses', {
           url: '/licenses',
-          templateProvider: ['$templateCache', function ($templateCache) {
-            return $templateCache.get('partials/purchase/purchase-licenses.html');
-          }],
-          controller: 'PurchaseLicensesCtrl',
+          abstract: true,
+          template: '<ui-view/>',
```
Comment (alex-deaconu) [inline web/scripts/purchase/app.js:63]: @santiagonoguez can you use a closing tag here? Thanks.

File: web/scripts/purchase/app.js [hunk 3]
```diff
@@ -36,9 +37,15 @@ angular.module('risevision.apps')
                 return canAccessApps()
                   .then(function () {
                     if (currentPlanFactory.isSubscribed() && !currentPlanFactory.isParentPlan()) {
-                      $state.go('apps.purchase.licenses', {
-                        displayCount: $stateParams.displayCount
-                      });
+                      if ($stateParams.purchaseAction === 'remove') {
```
Comment (alex-deaconu) [inline web/scripts/purchase/app.js:40]: @santiagonoguez I don't think this is necessary. We should just be able to call `apps.purchase.licenses.remove` directly. This usecase is very specific to scenarios regarding purchase/adding licenses. Thanks.

File: web/scripts/purchase/app.js [hunk 4]
```diff
@@ -83,10 +88,33 @@ angular.module('risevision.apps')
...
                   });
               }
-            ],
+            ]
+          }
+        })
+        .state('apps.purchase.licenses.add', {
+          url: '/add',
+          templateProvider: ['$templateCache', function ($templateCache) {
+            return $templateCache.get('partials/purchase/purchase-licenses-add.html');
```
Comment (alex-deaconu) [inline web/scripts/purchase/app.js:97]: @santiagonoguez maybe we can simplify and call this `add-licenses.html` and subsequently `remove-licenses.html`? Thanks.

File: web/partials/purchase/purchase-licenses-remove.html (hunk 1 above)
Comment (alex-deaconu) [reply web/partials/purchase/purchase-licenses-remove.html:6]: @santiagonoguez I would re-name loaders for this and the add page to reflect the name of the partial. In this case `purchase-licenses-remove-loader` or `remove-licenses-loader` if renamed as per the below. Thanks.

**Review by alex-deaconu — COMMENTED** [review]
@santiagonoguez looks good, few suggestions inline. Thanks.

File: web/partials/purchase/purchase-licenses-remove.html (hunk 1 above)
Comment (santiagonoguez) [reply web/partials/purchase/purchase-licenses-remove.html]: @alex-deaconu spinner and pages reloaded. I also created a different controller so each one deals with its own spinner and validation logic as @ezequielc suggested before, thanks !

File: web/scripts/purchase/app.js (hunk 2 above)
Comment (santiagonoguez) [reply web/scripts/purchase/app.js]: Updated. I copied the ui-view style from the rest of apps so it's consistent now, thanks !

File: web/scripts/purchase/app.js (hunk 3 above)
Comment (santiagonoguez) [reply web/scripts/purchase/app.js]: Reverted, thanks !

**Comment by santiagonoguez** [comment]
@alex-deaconu all suggestions applied. I'll create additional branchs on top of this for the next round of changes / implementation, thanks !

**Comment by ezequielc** [comment]
Looks good to me 👍

**Review by alex-deaconu — APPROVED** [review]
@santiagonoguez looks good, thanks for the updates. Thinking if it's worth duplicating Add/Remove partial and controller or not. Thanks.

**Comment by santiagonoguez** [comment]
@ezequielc @alex-deaconu I'm merging this to master now, if you are ok with that, thanks !

**Review by alex-deaconu — APPROVED** [review]
@santiagonoguez looks good, please ensure you trim some of the merge commit ;). Thanks.

**Review by ezequielc — APPROVED** [review]
@santiagonoguez Good with me as well 👍  Thanks!

**Pull request merged**